pkg/clusterpoolhost/clusterclaim.go

```diff
@@ -224,29 +225,33 @@ func checkClusterClaimsRunning(dynamicClient dynamic.Interface, clusterClaimName
 		running := false
 		if len(cc.Spec.Namespace) != 0 {
 			cdu, err := dynamicClient.Resource(helpers.GvrCD).Namespace(cc.Spec.Namespace).Get(context.TODO(), cc.Spec.Namespace, metav1.GetOptions{})
-			if err != nil {
-				allErrors[clusterClaimName] = err
-				fmt.Printf("Error: %s\n", err.Error())
-				continue
-			}
-			cd := &hivev1.ClusterDeployment{}
-			if runtime.DefaultUnstructuredConverter.FromUnstructured(cdu.UnstructuredContent(), cd); err != nil {
-				allErrors[clusterClaimName] = err
-				fmt.Printf("Error: %s\n", err.Error())
-				continue
-			}
-			if cd.Spec.PowerState == hivev1.HibernatingClusterPowerState {
-				allErrors[clusterClaimName] = fmt.Errorf("%s is hibernating, run a use command to resume it", cc.GetName())
-				fmt.Printf("Error: %s\n", allErrors[clusterClaimName])
-				continue
-			}
-			c := getClusterClaimRunningStatus(cc)
-			if len(cd.Spec.ClusterMetadata.AdminPasswordSecretRef.Name) != 0 &&
-				len(cd.Spec.BaseDomain) != 0 &&
-				len(cd.Status.APIURL) != 0 &&
-				c != nil && c.Status == corev1.ConditionStatus(metav1.ConditionTrue) {
-				running = true
-				fmt.Printf("clusterclaim %s is running with id %s (%d/%d)\n", clusterClaimName, cc.Spec.Namespace, i, timeout)
+			if statusError, isStatus := err.(*errors.StatusError); isStatus && statusError.Status().Reason == metav1.StatusReasonForbidden {
+				fmt.Printf("Permissions error when accessing claimed ClusterDeployment.  Permissions are likely still propagating. \nError: %s\n", err.Error())
+			} else {
```

Comment:
yes I think we can skip the other tests

Reply:
Hmm, if we add:
```
allErrors[clusterClaimName] = err
```
we see the following behavior:
```
clusterclaim gurnben-test created
Permissions error when accessing claimed ClusterDeployment.  Permissions are likely still propagating. 
Error: clusterdeployments.hive.openshift.io "canary-aws-4719-cnh8j" is forbidden: User "system:serviceaccount:clusterpool:gurnben" cannot get resource "clusterdeployments" in API group "hive.openshift.io" in the namespace "canary-aws-4719-cnh8j"
clusterclaim gurnben-test is not running (1/60)
error: all requested clusterclaims have errors
```
and then the program exits.  

If we use continue instead of an if/else, one of the following steps writes out to allErrors and we see the same behavior.  

So I think we have to log the error but omit it from allErrors (because eventually the error is no longer observed and can be safely ignored) and we have to fuse off the other checks with an if/else, or else others will fire and cause a similar error.  

The current code-flow was the only way I could successfully ignore this error case that occurs in the first 1-3 poll steps.

Reply:
ok